core/src/io/anuke/mindustry/input/DesktopInput.java

```diff
@@ -267,6 +267,12 @@ void pollInput(){
         int cursorY = tileY(Core.input.mouseY());
         int rawCursorX = world.toTile(Core.input.mouseWorld().x), rawCursorY = world.toTile(Core.input.mouseWorld().y);
 
+        // automatically pause building if the current build queue is empty
+        if (Core.settings.getBool("buildautopause") && player.isBuilding && !player.isBuilding()) {
```

Comment:
Minor formatting issue: `if () {` -> `if(){`. Similarly, there's an issue in `BuildBlock`.